almond/Dockerfile

```diff
@@ -5,50 +5,77 @@ FROM ${BUILD_FROM}
 ENV \
     LANG="en_US.utf8" \
     THINGENGINE_HOME="/data/almond-server" \
-    THINGENGINE_HOST_BASED_AUTHENTICATION="local-ip"
+    THINGENGINE_HAS_REVERSE_PROXY=true \
+    THINGENGINE_HOST_BASED_AUTHENTICATION=insecure
 
 WORKDIR /opt/almond
 ARG ALMOND_VERSION
 
+RUN adduser --disabled-password almond && \
+   chown almond:almond /opt/almond
```

Comment:
We generally don't run under a different user. This might cause issues with things like restoring snapshots, as the permissions on those are not guaranteed. Have you taken care of that? If so, how?

Reply:
It's unsafe to run as root, especially cause you're not using rootless docker. It seems pretty obvious to me.

Reply:
That didn't answer my questions.

Reply:
I don't know if snapshots work with a separate user (although I expect they will if the user ID doesn't change from one image to the next, which it shouldn't).

Now, reading the docker more carefully, the user is not actually used as runtime. It is only used at build time. I didn't write this part of the docker file originally, but I expect it has to do with npm which is broken when run as root. In that case, snapshots should be fine because files in the data file system will be owned by root. Is it ok to leave the non-root user here then?

Reply:
If the user isn't used ad runtime, this stuff all isn't really needed and obsolete to begin with.
Please remove it.

Reply:
Again, I think it *is* needed because npm does not run as root, and the build will not run correctly without this. I can test if you insist - but please keep in mind we went through a lot of iterations to make this work, so we have learned what it takes.

Reply:
As said, we use it with Z-Wave JS without issue. For my personal add-ons; Z-Wave 2 MQTT, NGinx Proxy Manager, Jupyter Lab (extensions), Node-RED, Home Panel.

So I'm not sure why it shouldn't work here?

Reply:
Is it npm 6? And do those packages require "prepare" steps? In my experience, npm 6 will not run "prepare" steps when it is run as root, which results in a botched build.

Reply:
Now, reading through upstream node issues (https://github.com/npm/cli/issues/2062 https://github.com/npm/cli/issues/1287 https://github.com/npm/cli/issues/1214 and others) it seems the bug only occurs with git dependencies. The released version of Almond has no git dependencies, so it's possible that we can remove the non-root user here. Development versions of Almond do have git dependencies though, and I'd rather keep the docker build as close as possible between development and stable versions, so we don't introduce bugs when we build the stable release.

Reply:
Yes NPM 6.14+ on them all, including native extensions build with gyp